### PR TITLE
fix(xml): extract tags map to separate entry point for mobile & web

### PR DESCRIPTION
**Pull request by joshuayoes**
# Summary

**What issues does the pull request solve? Please tag them so that they will get automatically closed once the PR is merged**

When utilizing with `react-native-web`: SvgXml and SvgUri are not exported in the `src/ReactNativeSVG.web.ts` extension. The logic for SvgXml and SvgUri do not have native dependencies, so they are safe to consume on web.

Issues:
- https://github.com/software-mansion/react-native-svg/issues/1279
- https://github.com/software-mansion/react-native-svg/issues/1742

**What is the feature? (if applicable)**

Add SvgXml and SvgUri as consumable exports for `react-native-web`

**How did you implement the solution?**

Extract `tags` to shapes map into separate `tags.tsx` file, where native shape elements and web shape elements can be provided to their respective envs. 

**What areas of the library does it impact?**

`src` directory

## Test Plan

Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

### What's required for testing (prerequisites)?
n/a

### What are the steps to reproduce (after prerequisites)?
n/a

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅     |
| Android |    ✅     |

## Checklist

- [x] I have tested this on a device and a simulator
- [x] I added documentation in `README.md`
- [x] I updated the typed files (typescript)
- [ ] I added a test for the API in the `__tests__` folder

### CODE REVIEW

**Comment by joshuayoes** [comment]
Another way I considered going about this was creating `web.tsx` shadows for each exported element in the `src/elements` directory. 
Screen Shot 2022-11-09 at 1 02 29 PM

This would be nice because it would allow `src/ReactNativeSVG.web.ts` to more closely resemble `src/ReactNativeSVG.ts`. But I figured it would make the PR noisier with lots of smaller file changes.

**Comment by WoLewicki** [comment]
@joshuayoes thanks for the contribution! Do you have knowledge how does it go with the discussion from #1279? Especially security concerns e.g. from https://github.com/software-mansion/react-native-svg/issues/1279#issuecomment-595269518?

**Comment by joshuayoes** [comment]
> @joshuayoes thanks for the contribution! Do you have knowledge how does it go with the discussion from #1279? Especially security concerns e.g. from https://github.com/software-mansion/react-native-svg/issues/1279#issuecomment-595269518?

Thanks for your response. Taking a deeper look at those issues, I think this PR may not be a good solution for maximal security, which it seems the team has decided to prioritize.

However, it would be nice if there was a clearer error when trying to import these on the web. Perhaps there could be a web export that just throws an error with a link to the PRs?

**Comment by WoLewicki** [comment]
Yeah, it seems like a good solution. Maybe even better if there was a link to the #1279 and a note that if you want to use it, then there could be a link to the PRs. Would you mind opening such a PR?

**Comment by bohdanprog** [comment]
Hello @joshuayoes,
Would you be able to resolve the conflicts, please?

File: src/xml.tsx [hunk 1]
```diff
@@ -388,7 +339,7 @@ export function parse(source: string, middleware?: Middleware): JsxAST | null {
       props,
       children: [],
       parent: currentElement,
-      Tag: tags[tag] || missingTag,
+      Tag: getTag(tag as keyof typeof tags),
```
Comment (jakex7) [inline src/xml.tsx:342]: Why do we need this change? Shouldn't tags from `tags.tsx` and `tags.web.tsx` be exactly the same?

File: src/xml.tsx [hunk 2]
```diff
@@ -485,6 +436,16 @@ export function parse(source: string, middleware?: Middleware): JsxAST | null {
     return name;
   }
 
+  function getTag(
+    name: keyof typeof tags
+  ): ComponentType<{ children?: React.ReactNode }> {
```
Comment (jakex7) [inline src/xml.tsx:441]: I don't think that `ComponentType<{ children?: React.ReactNode }>` is the right type for Svg tags, i.e. `Circle` or `Rect`

File: src/xml.tsx [hunk 3]
```diff
@@ -485,6 +436,16 @@ export function parse(source: string, middleware?: Middleware): JsxAST | null {
...
+    type Element = (typeof tags)[keyof typeof tags];
+    const Tag: Element | undefined = tags[name];
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    return Tag || missingTag;
```
Comment (jakex7) [inline src/xml.tsx:446]: That TS error is valid and should not be muted. We should correct return type of `getTag` function instead.

Comment (bohdanprog) [reply src/xml.tsx]: fixed.

File: src/xml.tsx (hunk 2 above)
Comment (bohdanprog) [reply src/xml.tsx]: removed that function.

File: src/xml.tsx (hunk 1 above)
Comment (bohdanprog) [reply src/xml.tsx]: changed that, and restored the old version.

**Review by jakex7 — APPROVED** [review]
LGTM, thank you! 🚀

**Pull request merged**